test/modules/aws-opensearch-integration.ts

```diff
@@ -0,0 +1,206 @@
+                              enable_fine_grained_access_control, fine_grained_access_control_master_username,
+                              fine_grained_access_control_master_password, access_policy, region)
+          VALUES ('${prefix}', 'OpenSearch_2.3', 1, 'c5.large.search', 1,
+                  '{"Iops": 3000, "EBSEnabled": true, "Throughput": 125, "VolumeSize": 10, "VolumeType": "gp3"}', true,
+                  'admin', '${initialPassword}',
+                  '{"Version": "2012-10-17", "Statement": [{"Action": "es:*", "Effect": "Allow", "Resource": "*", "Principal": {"AWS": "*"}}]}',
+                  '${region}');
+      `,
+      undefined,
+      true,
+      () => ({ username, password }),
+    ),
+  );
+  it('applies creation of the domain and waits for the creation', commit());
+
+  itDocs(
+    'gets the domain and fetches it using the username and password',
+    query(
+      `
+    SELECT endpoint FROM domain WHERE domain_name = '${prefix}';
+  `,
+      (res: any[]) => {
+        expect(res.length).toBe(1);
+
+        const headers: Record<string, string> = {};
+        headers.Authorization = 'Basic ' + Buffer.from('admin:' + initialPassword).toString('base64');
+        fetch('https://' + res[0].endpoint, {
+          method: 'GET',
+          headers,
+        })
+          .then((r: any) => r.json())
+          .then((r: any) => expect(r.version.number).toBe('2.3.0'));
+
+        // should not respond with wrong password
+        headers.Authorization = 'Basic ' + Buffer.from('admin:wrongpass').toString('base64');
+        fetch('https://' + res[0].endpoint, {
+          method: 'GET',
+          headers,
+        })
+          .then((r: any) => r.text())
+          .then((r: string) => expect(r).toBe('Unauthorized'));
+      },
+    ),
+  );
+
+  it('starts a transaction', begin());
+  itDocs(
+    'changes the password of the domain',
+    query(
+      `
+          UPDATE domain
+          SET fine_grained_access_control_master_username = 'admin',
+              fine_grained_access_control_master_password = '${secondPassword}'
+          WHERE domain_name = '${prefix}';
+      `,
+      undefined,
+      true,
+      () => ({ username, password }),
+    ),
+  );
+  it('applies change of the password for the domain', commit());
+
+  it(
+    'fetches the domain using the second password',
+    query(
+      `
+    SELECT endpoint FROM domain WHERE domain_name = '${prefix}';
+  `,
+      (res: any[]) => {
+        expect(res.length).toBe(1);
+
+        const headers: Record<string, string> = {};
+        headers.Authorization = 'Basic ' + Buffer.from('admin:' + secondPassword).toString('base64');
+        fetch('https://' + res[0].endpoint, {
+          method: 'GET',
+          headers,
+        })
+          .then((r: any) => r.json())
+          .then((r: any) => expect(r.version.number).toBe('2.3.0'));
+      },
+    ),
+  );
+
```

Comment:
can we add a test here uninstalling and then installing again the module and check if the `domain` syncs properly from cloud?